edit_tiles/app.py

```
@@ -7616,6 +7926,25 @@ def _next_town_layer_color(self):
+                self.mod_project.sources.index(active_source)
+                if active_source in self.mod_project.sources else None
+            )
+            for index, layer in enumerate(self.mod_project.layers):
+                if (layer.is_fuse_native
+                        and not layer.read_only
+                        and (source_idx is None
+                             or getattr(layer, "source_idx", None) == source_idx)):
+                    return index, layer
+            raise RuntimeError(
+                "The native FUSE package has no editable FuseDataFiles layer"
+            )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)